File: app/src/main/java/ru/ulstu_team/ulstuschedule/injection/component/StudentScheduleComponent.java
package ru.ulstu_team.ulstuschedule.injection.component;

import dagger.Component;
import ru.ulstu_team.ulstuschedule.injection.PerActivity;
import ru.ulstu_team.ulstuschedule.injection.module.StudentScheduleModule;
import ru.ulstu_team.ulstuschedule.ui.common.student.StudentScheduleFragment;

@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = StudentScheduleModule.class)
public interface StudentScheduleComponent {

    void inject(StudentScheduleFragment scheduleFragment);

}

File: app/src/main/java/ru/ulstu_team/ulstuschedule/injection/component/StudentScheduleComponent.kt
package ru.ulstu_team.ulstuschedule.injection.component

import dagger.Component
import ru.ulstu_team.ulstuschedule.injection.PerActivity
import ru.ulstu_team.ulstuschedule.injection.module.StudentScheduleModule
import ru.ulstu_team.ulstuschedule.ui.common.group.GroupScheduleFragment

@PerActivity
@Component(dependencies = arrayOf(ApplicationComponent::class), modules = arrayOf(StudentScheduleModule::class))
interface StudentScheduleComponent {

    fun inject(scheduleFragment: GroupScheduleFragment)

}
